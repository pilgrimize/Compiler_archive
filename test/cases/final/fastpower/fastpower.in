43 79 589432