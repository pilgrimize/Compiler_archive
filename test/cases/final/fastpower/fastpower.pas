program main;
var a, b, c:longint;
function fpow(x, y, m:longint):longint;
var tmp:integer;
begin
  if y=0 then
  begin
    fpow:=1;
  end
  else begin
    tmp:=fpow(x, y div 2, m);
    fpow:=(tmp*tmp) mod m;
    if y mod 2=1 then
      fpow:=fpow*x mod m;
  end;
end;
begin
  read(a,b,c);
  writeln(a,' ^ ',b,' mod ',c,' = ',fpow(a,b,c));
end.