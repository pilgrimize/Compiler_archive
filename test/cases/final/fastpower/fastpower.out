43 ^ 79 mod 589432 = 45227
